kolibri/core/kolibri_plugin.py

```diff
@@ -14,15 +14,22 @@
 from kolibri.plugins.hooks import NAVIGATION_POPULATE
 
 
-class KolibriCore(KolibriPluginBase):
+def main_navigation():
+    """
+    Callback for the plugin below
+    :return: A list of nav menu items used in the NAVIGATION_POPULATE hook.
+    """
+    return [{
+        'menu_name': _("Start page"),
+        'menu_url': reverse('kolibri:index'),
+    }]
 
-    def main_navigation(self):
-        return [{
-            'menu_name': _("Start page"),
-            'menu_url': reverse('kolibri:index'),
-        }]
 
-    def hooks(self):
-        return {
-            NAVIGATION_POPULATE: self.main_navigation
-        }
+class KolibriCore(KolibriPluginBase):
+    """
+    The most minimal plugin possible. Because it's in the core, it doesn't define ``enable`` or ``disable``. Those
+    methods should never be called for this plugin.
+    """
+    hooks = {
```

Comment:
I have to disagree with this version... if it's a callable, it gives the plugin the power to do stuff more conditionally. Like defining which hooks it's going to handle and which callables it wants to give them... this assignment is static and happens and load time.